src/main/java/uk/gov/companieshouse/api/accounts/controller/ApprovalController.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.model.rest.Approval;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.impl.ApprovalService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.logging.util.LogContext;
+import uk.gov.companieshouse.logging.util.LogHelper;
+
+@RestController
+@RequestMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/approval", produces = MediaType.APPLICATION_JSON_VALUE)
+public class ApprovalController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+    private static final String REQUEST_ID = "X-Request-Id";
+
+    @Autowired
+    private ApprovalService approvalService;
+
+    @Autowired
+    private ErrorMapper errorMapper;
+
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
+    @PostMapping
+    public ResponseEntity create(@RequestBody @Valid Approval approval,
+        BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+        HttpServletRequest request) {
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult, errors);
+
+        }
+
+        if (errors.hasErrors()) {
+
+            LOGGER.error(
+                "Approval validation failure");
+            logValidationFailureError(getRequestId(request), errors);
+
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+
+        }
+
+        Transaction transaction = (Transaction) request
+            .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        String requestId = request.getHeader(REQUEST_ID);
+
+        ResponseEntity responseEntity;
+
+        try {
+            ResponseObject<Approval> responseObject = approvalService
+                .create(approval, transaction, companyAccountId, requestId);
+            responseEntity = apiResponseMapper
```

Comment:
We could `return` here and in `line 92` so that we could get rid of the `ResponseEntity` variable. 
But I am happy if you want to use one exit point approach